oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/segment/SegmentRecord.java

```diff
@@ -119,7 +121,9 @@ public String id() {
             map.put(TRACE_ID, storageData.getTraceId());
             map.put(SERVICE_ID, storageData.getServiceId());
             map.put(SERVICE_INSTANCE_ID, storageData.getServiceInstanceId());
-            map.put(ENDPOINT_NAME, storageData.getEndpointName());
+            if (nonNull(storageData.getEndpointName())) {
+                map.put(ENDPOINT_NAME, storageData.getEndpointName());
+            }
             map.put(ENDPOINT_ID, storageData.getEndpointId());
```

Comment:
I think this should not be added. Once this has error, the code already has bug. Trace should not have no endpoint.